Sources/SwiftGodot/Core/SignalRegistration.swift

```diff
@@ -0,0 +1,392 @@
+//
+//  SignalRegistration.swift
+//
+//
+//  Created by Padraig O Cinneide on 2023-10-18.
+//
+
+/// Describes a signal and its arguments.
+/// - note: It is recommended to use the #signal macro instead of using this directly.
+public struct SignalWithNoArguments {
```

Comment:
I'm noticing that `SignalWithNArguments` feels ergonomically similar to that of SwiftUI's TupleView. Perhaps we can glean from that, so that we're not manually writing all these cases out?

Reply:
On the surface, I don't see a way that can work and still give us the nice `emit` functions that enforce the types, but if you can say more about what you have in mind I can take another look.